### PR TITLE
DATAES-208 Allow node configuration to honor the server settings in @Document

### PR DESCRIPTION
**Pull request by dandonovsc**

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java [hunk 1]
```diff
@@ -879,6 +880,10 @@ public boolean createIndex(String indexName, Object settings) {
 	}
 
 	private <T> Map getDefaultSettings(ElasticsearchPersistentEntity<T> persistentEntity) {
+
+        if (persistentEntity.isServerConfig())
+            return Maps.newHashMap();
```
Comment (martin-g) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java]: Mixed whitespaces introduced.

File: src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java [hunk 2]
```diff
@@ -70,6 +71,7 @@ public SimpleElasticsearchPersistentEntity(TypeInformation<T> typeInformation) {
 					" Unknown indexName. Make sure the indexName is defined. e.g @Document(indexName=\"foo\")");
 			this.indexName = typeInformation.getType().getAnnotation(Document.class).indexName();
 			this.indexType = hasText(document.type()) ? document.type() : clazz.getSimpleName().toLowerCase(Locale.ENGLISH);
+			this.serverConfig = typeInformation.getType().getAnnotation(Document.class).serverConfig();
 			this.shards = typeInformation.getType().getAnnotation(Document.class).shards();
 			this.replicas = typeInformation.getType().getAnnotation(Document.class).replicas();
 			this.refreshInterval = typeInformation.getType().getAnnotation(Document.class).refreshInterval();
```
Comment (martin-g) [inline src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java]: These `typeInformation.getType().getAnnotation(Document.class)` could be cached in a local variable. No need to re-read several times.

**Comment by martin-g** [comment]
The issue title refers to DATAES-207 while it actually is https://jira.spring.io/browse/DATAES-208

**Comment by dandonovsc** [comment]
@martin-g  this is correct. I opened a few pull requests and somehow github crossed wires. Should I reopen with the correct subject , or is this ok for merging the code and changing the commit message?

**Comment by martin-g** [comment]
I think you should be able to 
1) edit the title of the PR 
2) rebase your local branch and push again.

File: src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java (hunk 2 above)
Comment (dandonovsc) [inline src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java]: Agreed. Sounds like a good chance for some code clean up

**Comment by dandonovsc** [comment]
The suggested changes have been applied. Is there another blocker for this pull request?

**Comment by akonczak** [comment]
Hi @dandonovsc 
Thank you for your contribution
I did rebase and squash on your code to have only one commit [branch/DATAES-208](https://github.com/spring-projects/spring-data-elasticsearch/tree/DATAES-208). 

Also I would like to make some changes to your code. 
1. rename serverConfig@Document to useServerConfiguration 
2. change default value to false so that current behaviour will backward compatible



Can you sign [contributor agreement](https://support.springsource.com/spring_committer_signup) ?  (Please ignore this request if you did it earlier)

**Comment by dandonovsc** [comment]
Hi @akonczak ,

I am totally fine with the changes you're going to make. 

I have signed the agreement already 

Thanks

**Pull request closed without merging**